tool/gravity/cli/system.go

```diff
@@ -672,11 +672,21 @@ func systemUninstall(env *localenv.LocalEnvironment, confirmed bool) error {
 	env.Backend.Close()
 
 	logger := log.WithField(trace.Component, "system:uninstall")
-	if err := environ.UninstallServices(env, logger); err != nil {
-		log.WithError(err).Warn("Failed to uninstall agent services.")
+	logger.Info("Uninstall system package services.")
+	if err := environ.UninstallPackageServices(env, logger); err != nil {
+		logger.WithError(err).Warn("Failed to uninstall package services.")
+		// Do not attempt to remove state if a package service failed to uninstall since
+		// this might result in inability to re-attempt uninstallation after fixing the issue.
+		// Instead bail out early and output the relevant error message
+		return trace.Wrap(err)
 	}
+	logger.Info("Uninstall system.")
 	if err := environ.UninstallSystem(context.TODO(), env, logger); err != nil {
-		log.WithError(err).Warn("Failed to uninstall system.")
+		logger.WithError(err).Warn("Failed to uninstall system.")
```

Comment:
I wonder if this and "uninstall agent services" below should also return an error if they fail, like you did above. Otherwise our uninstall is not reliable as users may rely on its successful exit code as indicator that everything was cleaned up while it in fact may not be true. Can't remember why it was done like this in the first place...